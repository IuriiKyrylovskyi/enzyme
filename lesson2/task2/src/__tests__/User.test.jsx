import React from "react";
import { shallow } from "enzyme";
import User from "../User";
import { getUserData } from "../usersGateway";

jest.mock("../usersGateway", () => {
  return {
    getUserData: jest.fn(() => Promise.resolve()),
  };
});

describe("User", () => {
  it("should not render without data", () => {
    const wrappedComponent = shallow(<User />);

    expect(wrappedComponent.find(".user").exists()).toEqual(false);
  });

  it("should render user with data", () => {
    const wrappedComponent = shallow(<User />);
    wrappedComponent.setState({
      userData: {
        avatar_url: "https://join.com",
        name: "John",
        location: "Kyiv",
      },
    });

    expect(wrappedComponent.find(".user").exists()).toBeTruthy();
  });

  it("should render user with data for snapshot", () => {
    const wrappedComponent = shallow(<User />);
    wrappedComponent.setState({
      userData: {
        avatar_url: "https://join.com",
        name: "John",
        location: "Kyiv",
      },
    });

    expect(wrappedComponent.find(".user")).toMatchSnapshot();
  });

  it("should call getUserData", () => {
    shallow(<User />);

    expect(getUserData).toBeCalled();
  });
});
