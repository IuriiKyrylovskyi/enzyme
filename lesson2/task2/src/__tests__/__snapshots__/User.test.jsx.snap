// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render user with data for snapshot 1`] = `
<div
  className="user"
>
  <img
    alt="User Avatar"
    className="user__avatar"
    src="https://join.com"
  />
  <div
    className="user__info"
  >
    <span
      className="user__name"
    >
      John
    </span>
    <span
      className="user__location"
    >
      Kyiv
    </span>
  </div>
</div>
`;
